source/backend/api/Services/ManagementFileService.cs

```diff
@@ -249,6 +252,39 @@ public Paged<PimsManagementFile> GetPage(ManagementFilter filter)
             return _managementFileRepository.GetPageDeep(filter);
         }
 
+        public PimsPropertyActivity GetActivity(long managementFileId, long activityId)
+        {
+            _logger.LogInformation("Retrieving property Activity with Id: {ActivityId}", activityId);
+            _user.ThrowIfNotAuthorized(Permissions.ManagementView);
+            _user.ThrowIfNotAuthorized(Permissions.PropertyView);
+
+            var propertyActivity = _propertyActivityRepository.GetActivity(activityId);
+
+            if (propertyActivity.ManagementFileId.HasValue && propertyActivity.ManagementFileId == managementFileId)
+            {
+                return propertyActivity;
+            }
+
+            throw new BadRequestException("Activity with the given id does not match the management file id");
+        }
+
+        public PimsPropertyActivity UpdateActivity(long managementFileId, long activityId, PimsPropertyActivity propertyActivity)
+        {
+            _logger.LogInformation("Updating property Activity with Id: {ActivityId}", activityId);
+            _user.ThrowIfNotAuthorized(Permissions.ManagementEdit);
+            _user.ThrowIfNotAuthorized(Permissions.PropertyEdit);
+
+            if (!propertyActivity.ManagementFileId.HasValue || propertyActivity.ManagementFileId != managementFileId || propertyActivity.Internal_Id != activityId)
```

Comment:
The managementFileId value is not used in the service call. If it is only being used for validation, let's move it to the controller given that one is responsible of validating the request.

Reply:
This is following the same patterns as **PropertyService.UpdateActivity()** for ad-hoc activities. I still see value in validating inputs at the service level to make the code more defensive (in the same way that we validate user permissions at the service level instead at the controller level). But I would be okay moving it to the controller id others feel like we should do that. Let's discuss this in our dev meeting later...

![image](https://github.com/user-attachments/assets/cfcc2630-d22b-43cf-98e2-48ff1fe34e8e)

Reply:
Yes, we should talk about this.